Code/Tools/FBuild/FBuildCore/BFF/Functions/FunctionDependencyList.cpp

```diff
@@ -0,0 +1,104 @@
+// FunctionAlias
+//------------------------------------------------------------------------------
+
+// Includes
+//------------------------------------------------------------------------------
+#include "FunctionDependencyList.h"
+#include "Tools/FBuild/FBuildCore/FBuild.h"
+#include "Tools/FBuild/FBuildCore/FLog.h"
+#include "Tools/FBuild/FBuildCore/BFF/BFFStackFrame.h"
+#include "Tools/FBuild/FBuildCore/BFF/BFFVariable.h"
+#include "Tools/FBuild/FBuildCore/Graph/AliasNode.h"
+#include "Tools/FBuild/FBuildCore/Graph/DependencyListNode.h"
+#include "Tools/FBuild/FBuildCore/Graph/NodeGraph.h"
+
+// Core
+#include "Core/FileIO/PathUtils.h"
+
+// CONSTRUCTOR
+//------------------------------------------------------------------------------
+FunctionDependencyList::FunctionDependencyList()
+: Function( "DependencyList" )
+{
+}
+
+// AcceptsHeader
+//------------------------------------------------------------------------------
+/*virtual*/ bool FunctionDependencyList::AcceptsHeader() const
+{
+    return true;
+}
+
+// Commit
+//------------------------------------------------------------------------------
+/*virtual*/ bool FunctionDependencyList::Commit( NodeGraph & nodeGraph, const BFFToken * funcStartIter ) const
+{
```

Comment:
You don't need any of this logic.  With the correct MetaData on the Node, it's all taken care of.
You only need this:
{code}
// CreateNode
//------------------------------------------------------------------------------
/*virtual*/ Node * FunctionDependencyList::CreateNode() const
{
    return FNEW( DependencyListNode );
}
{code}

See FunctionTextFile/TextFileNode for an example: